addons/website_slides/models/slide_channel.py

```diff
@@ -757,6 +877,33 @@ def _action_request_access(self, partner):
                 )
         return activities
 
+    def _get_channel_invitation_link(self, partner_id=False):
+        ''' This will return the url used as:
+            - (1) course sharing link displayed in the invitation wizard.
+            - (2) hyperlink in the channel invitation email in template mail_notification_channel_invite.
+
+        Corresponding invitation links :
+        (1) : always returns the default /invite route
+        (2) - if partner_id has no user :
+                - if free sign up is allowed and the course has public visibility, then the user can browse
+                  freely the course and sign up only if desired. Therefore add auth_signup_token in url, kept
+                  in session and linking partner_id to created user on sign up.
+                - if no free sign up is allowed or the course is visible to members only, then the partner must
+                  directly receive a signup link, redirecting to the /invite route
+
+        :param partner_id: (optional) partner id of the invited partner, in direct invitation emails.
+        :return: relevant channel invitation url, see above.
+        '''
+        url = '%s/slides/%s/invite?' % (self.get_base_url(), self.id)
+        if partner_id and partner_id.partner_share:
+            signup_allowed = self.env['res.users'].sudo()._get_signup_invitation_scope() == 'b2c'
+            partner_id.sudo().signup_prepare()
+            if self.visibility == 'members' or not signup_allowed:
```

Comment:
The idea was that a partner without user invited to a visible course would not need an account creation link if it is able to create its account itself (if Free sign up is on).

For instance I invite Roger to "on invite" / "visible to anyone" course. We wanted the user to be able to browse the course before deciding to create an account and join it, hence this extra check : 
-> "Can the partner create its own account? Is free sign up on ?"
Otherwise it will need to be invited... Therefore if Free signup is off, we directly send an account creation link so that the user can create it then access the course directly if he wants to.

If free signup is on, then it can browse the course then create an account if desired, using the auth_signup token to link the partner to this new user and prefill the new portal user form.